pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -245,6 +247,10 @@ func (rc *reconciler) reconcile() {
 
 	rc.attachDesiredVolumes()
 
+	// Update metrics
+	metrics.RecordADControllerVolumesMetric("actual_state_of_world", len(rc.actualStateOfWorld.GetAttachedVolumes()))
+	metrics.RecordADControllerVolumesMetric("desired_state_of_world", len(rc.desiredStateOfWorld.GetVolumesToAttach()))
+
```

Comment:
I would remove this from here because there is a certain cost associated with recording a metric and the reconciler loop IMO turns too quickly(100ms) while metric collection will happen every 10s. So prometheus client library will "aggregate" these metrics client side.

So we should consider introducing another timer for collecting these metrics rather than using reconciler.

Reply:
Done! Thanks for the heads up!